src/document/proxy/array_proxy.ts

```diff
@@ -290,6 +300,23 @@ export class ArrayProxy {
     }
   }
 
+  /**
+   * `insertBeforeInternal` inserts the value before the previously created element.
+   */
+  public static insertBeforeInternal(
+    context: ChangeContext,
+    target: JSONArray,
+    nextCreatedAt: TimeTicket,
+    value: unknown,
+  ): JSONElement {
+    return ArrayProxy.insertAfterInternal(
+      context,
+      target,
+      target.getPrevCreatedAt(nextCreatedAt),
```

Comment:
`target.getPrevCreatedAt` finds the previous node and not deleted.

If the node discovered by `target.getPrevCreatedAt` is `dummyHead`, it is added after it.